balancer/endpointsharding/endpointsharding_test.go

```diff
@@ -157,3 +157,26 @@ func (s) TestEndpointShardingBasic(t *testing.T) {
 		t.Fatalf("error in expected round robin: %v", err)
 	}
 }
+
+// TestEndpointShardingBalancerAPI tests that when endpoint sharding gets a
+// Client Conn update with a duplicate address across endpoints the
+// UpdateClientConnState operation errors.
+func (s) TestEndpointShardingBalancerAPI(t *testing.T) {
+	es := NewBalancer(nil, balancer.BuildOptions{})
+	addr := resolver.Address{Addr: "addr1"}
+	err := es.UpdateClientConnState(balancer.ClientConnState{
+		ResolverState: resolver.State{
```

Comment:
This is pretty limited in what it covers.  E.g. it doesn't cover addresses across endpoints, or endpoints with multiple addresses.  Please add some more test cases to cover more possibilities -- probably make it table driven where `[]Endpoint` is the input and the result is probably just a `bool` to indicate whether there should be an error or not.

Reply:
Added more coverage for passing and failing tests.